### ISSUE TITLE
Add User Manual

### ISSUE THREAD

**Issue by rhuss** [knative/docs#2810]
## Describe the feature:

The user manual should be use case focussed and guide the user through several important use cases. That's different to the reference manual which is more for a quick lookup of options.

The user manual should live on knative.dev, we just track the issue here (but could move it over, too)

I could imagine a TOC like in:

* Installation of kn
* Authentication and configuration
* Knative Serving
   - Create a simple service and verify that its running
   - What does `kn service create --force` ?
   - What is a BYO revision name and why it is useful ?
   - How to tune concurrency parameter ?
   - How to change requests/limits
   - How to add labels
   - How to update a service ?
   - What is a Traffic split ?
   - How can I perform a canary deployment ? A blue green deployment ?
   - How can I rollback ?
   - What does the output of `kn service describe` actually means, especially the output for the traffic split ?
   - What the difference between _latest created_ and _latest ready_ and how can I see this in the client ?

* Knative Eventing
   - Basic example with source
   - Which sources are supported ?
   - How to use triggers and what are the prerequisites ? (i.e. a broker that is enabled)
   - Which sink types are supported ?
    -....

* Plugins
   - What is a plugin and how can I implement it ?
   - Where to find plugins ? (client-contrib)
   - How to install a plugin and how does the lookup mechanism work ? How can this be changed/configured ?
   - How to list all plugins ?
   - How does a eventing source plugin looks like and how it is different to other plugins ? (i.e using a manifest)

The style of the user manual should be relaxed and easy to read. The section can build upon each other.

The User Manual is hosted as part of knative.dev

This issue is part of knative/client#510

**Comment by abrennan89** [comment]
Note to self: Related to https://github.com/knative/ux/issues/39

**Comment by abrennan89** [comment]
@daisy-ycguo @rhuss is anyone currently working on this or should we unassign it and split up the issue into smaller tasks?

**Comment by rhuss** [comment]
@abrennan89 no, nobody is actually working on this issue (an @daisy-ycguo is not working on Knative anymore ...)

**Comment by abrennan89** [comment]
I don't think that this should be one single guide for CLI, but let's see what comes out of Omer's card sorts in terms of docs structure.
I think this should've been filed as individual issues per topic, not one giant issue, but I've converted the list to checkboxes. I think in the meantime let's cross off what already exists in docs and fill any gaps, then the kn content in the site can be restructured later if it needs to be.

**Comment by abrennan89** [comment]
@snneji any thoughts on where this fits into the new card sort version of docs? I don't think we have a CLI section anymore but kn is a major part of Knative.

**Comment by abrennan89** [comment]
@psschwei it looked like you were working on some sort of automation for this, is that still the plan going forward? Not sure if there's any work to do here still?

**Comment by psschwei** [comment]
I just added a link to the client docs to the knative docs... it would be good to add it to the site, but it's not something I'd be able to work on in the near future...

**Comment by abrennan89** [comment]
Proposal for docs WG - how do we feel about adding top level CLI and Functions sections to the site, the same way we have for Serving and Eventing? More specifically, any arguments against doing this? To me they're the other two largest UX components that I could see folks wanting to find quickly. cc @psschwei @csantanapr @snneji

**Comment by abrennan89** [comment]
We have discussed this one in the Docs WG and although we have decided not to include the content in this exact format / layout, we plan to check that each of these requests are documented, or else maybe open individual issues for dealing with any gaps in the docs.

**Comment by abrennan89** [comment]
Plugins requirements split into separate issue: https://github.com/knative/docs/issues/3534

**Comment by rhuss** [comment]
@abrennan89 `knative service create --force` is to entirely overwrite an existing service, which means delete it first and then recreate it. This is in contrast to `kn service update`, which updates individual fields on an existing installation. You would use it in scripts to ensure that you have a fresh service even if a service with this name already exists. That's useful e.g. for CI runs when you are not sure that the previous run has been cleaned up properly (which can happen when a CI tests fails).

Also it's different to `kn service apply` as it does not merge on an existing service, that might have been changed by a third party. It just overwrites any changes that had happened. It's a sledgehammer.

**Comment by abrennan89** [comment]
Closing this since the issue has been open for over 2 years. @rhuss please do not open XXL issues like this, and instead open an issue for each individual section if there are changes still required. Opening issues that are this large make working them unmanageable.
